src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -263,16 +263,16 @@ void SyncManager::set_log_level(util::Logger::Level level) noexcept
     m_config.log_level = level;
 }
 
-void SyncManager::set_logger_factory(SyncLoggerFactory& factory) noexcept
+void SyncManager::set_logger_factory(std::function<SyncLoggerFactory> factory) noexcept
```

Comment:
Can this be std::function<SyncLoggerFactory()> to make it clear that it's a function?

Reply:
Looking at https://en.cppreference.com/w/cpp/utility/functional/function, this may actually be undefined?